pkg/collector/corechecks/snmp/checkconfig/config.go

```diff
@@ -581,30 +585,50 @@ func (c *CheckConfig) IsDiscovery() bool {
 	return c.Network != ""
 }
 
-func parseScalarOids(metrics []MetricsConfig, metricTags []MetricTagConfig) []string {
+func (c *CheckConfig) parseScalarOids(metrics []MetricsConfig, metricTags []MetricTagConfig, metadataConfigs MetadataConfig) []string {
 	var oids []string
 	for _, metric := range metrics {
-		if metric.Symbol.OID != "" {
-			oids = append(oids, metric.Symbol.OID)
-		}
+		oids = append(oids, metric.Symbol.OID)
 	}
 	for _, metricTag := range metricTags {
-		if metricTag.OID != "" {
-			oids = append(oids, metricTag.OID)
+		oids = append(oids, metricTag.OID)
+	}
+	if c.CollectDeviceMetadata {
+		for resource, metadataConfig := range metadataConfigs {
+			if !IsMetadataResourceWithScalarOids(resource) {
+				continue
+			}
+			for _, field := range metadataConfig.Fields {
+				oids = append(oids, field.Symbol.OID)
+			}
+			// we don't support tags for now for resource (e.g. device) based on scalar OIDs
+			// profile root level `metric_tags` (tags used for both metadata, metrics, service checks)
+			// can be used instead
 		}
 	}
 	return oids
 }
 
-func parseColumnOids(metrics []MetricsConfig) []string {
+func (c *CheckConfig) parseColumnOids(metrics []MetricsConfig, metadataConfigs MetadataConfig) []string {
 	var oids []string
 	for _, metric := range metrics {
 		for _, symbol := range metric.Symbols {
 			oids = append(oids, symbol.OID)
 		}
 		for _, metricTag := range metric.MetricTags {
-			if metricTag.Column.OID != "" {
-				oids = append(oids, metricTag.Column.OID)
+			oids = append(oids, metricTag.Column.OID)
+		}
+	}
+	if c.CollectDeviceMetadata {
+		for resource, metadataConfig := range metadataConfigs {
+			if IsMetadataResourceWithScalarOids(resource) {
+				continue
+			}
+			for _, field := range metadataConfig.Fields {
+				oids = append(oids, field.Symbol.OID)
+			}
+			for _, tagConfig := range metadataConfig.IDTags {
+				oids = append(oids, tagConfig.Column.OID)
```

Comment:
why columns are allowed for idtags? Does this mean that a device can have multiple idtags? What do idtags identify ?

Reply:
> why columns are allowed for idtags? 

idTags are define like this (see also _generic-if.yaml file in this PR)

```

metadata:
  interface:
    fields:
      name:
        symbol:
          OID: 1.3.6.1.2.1.31.1.1.1.1
          name: ifName
[...]
    id_tags:  # id_tags/tags are special fields
      - column:
          OID: 1.3.6.1.2.1.31.1.1.1.1
          name: ifName
        tag: interface
```

> Does this mean that a device can have multiple idtags? 

For now, idtags can be defined for resources based on column OIDs (`interface` yes, not for `device`)

> What do idtags identify ?

I'm not sure to understand. `id_tags` are used for correlation with metrics as before this PR, for interface, it correspond to `interface.id_tags` field in ES

Reply:
I got confused as I associated `idTags` to device `idTags`.

Reply:
let me know if something can be improved to make it more clear